ui-core/src/util/taskUtils.ts

```diff
@@ -43,24 +43,57 @@ export function isTaskAccessible(task: ParticipantTask, enrollee: Enrollee) {
   if (task.taskType === 'CONSENT' || task.status === 'COMPLETE') {
     return true
   }
-  const openConsents = enrollee.participantTasks
+
+  // do not consider removed surveys in this logic
+  const tasks = enrollee.participantTasks.filter(t => isTaskVisible(t))
+
+  const openConsents = tasks
     .filter(task => task.taskType === 'CONSENT' && task.status !== 'COMPLETE')
   if (openConsents.length) {
     return false
   }
-  const openRequiredTasks = enrollee.participantTasks.filter(task => task.blocksHub && task.status !== 'COMPLETE')
-    .sort((a, b) => a.taskOrder - b.taskOrder)
-  if (openRequiredTasks.length) {
-    return task.targetStableId === openRequiredTasks[0].targetStableId
+
+  const openRequiredTasks = tasks
+    .filter(task => task.blocksHub)
+    .sort((a, b) => a.taskOrder !== b.taskOrder
+      ? a.taskOrder - b.taskOrder
+      : (a.targetStableId || '').localeCompare(b.targetStableId || ''))
+  if (openRequiredTasks.length === 0) {
+    return true
+  }
+
+  // make sure the task before this task has at least one completed survey -
+  // even if the previous survey has something new/in-progress/removed, if it has
+  // a single completed response it should be considered done.
+  let lastTaskStableId = ''
+  let lastTaskIsComplete = true // first required task always accessible
+  for (let i = 0; i < openRequiredTasks.length; i++) {
+    if (task.targetStableId === openRequiredTasks[i].targetStableId) {
+      return lastTaskIsComplete
+    }
+
+    if (openRequiredTasks[i].targetStableId === lastTaskStableId) {
+      if (!lastTaskIsComplete) {
+        lastTaskIsComplete = openRequiredTasks[i].status === 'COMPLETE'
+      }
+    } else {
+      lastTaskStableId = openRequiredTasks[i].targetStableId || ''
+      lastTaskIsComplete = openRequiredTasks[i].status === 'COMPLETE'
+    }
```

Comment:
I think this logic is really hard to follow.  Try something like below-- the intermediate variable names will help explain what's happening
```
const openRequiredStableIds = _uniq(openRequiredTasks.map(task => task.stableId))
const neverCompletedStableIds = openRequiredStableIds.filter(stableId => !openRequiredTasks.some(task => task.targetStableId === stableId && task.status === 'COMPLETE'))
return neverCompletedStableIds.length === 0 ||  task.targetStableId = neverCompletedStableIds[0]
```

Reply:
ahh, that's much nicer!